### PR TITLE
Fix humann2 rule

### PR DESCRIPTION
**Pull request by tanaes**
This updates the HUMAnN2 rule to successfully execute in Oecophylla. 

@mortonjt @qiyunzhu 

Depends on #98

### CODE REVIEW

File: oecophylla/cli/launch.py [hunk 1]
```diff
@@ -86,67 +122,97 @@ def _create_dir(_path):
...
+    # Check to see if running with test data. If so, fill in defaults
+    # for relevant empty parameters.
+    if test:
+        input_dir, sample_sheet, params, envs, output_dir = _setup_test()
+    elif not output_dir:
+        raise IOError("Must provide output directory to run.")
+
+    # Check to see if config.yaml exists in output dir. If it does, warn
+    # and continue with execution
+    if os.path.exists(os.path.join(output_dir,'config.yaml')):
```
Comment (qiyunzhu) [inline oecophylla/cli/launch.py]: whitespace missing between `output_dir,` and `'config.yaml'`

File: oecophylla/cli/launch.py [hunk 2]
```diff
@@ -49,18 +49,54 @@ def _create_dir(_path):
...
+    params = os.path.join(d,
+                          'test_data/test_config/test_params.yml')
+    envs = os.path.join(d,
+                        'test_data/test_config/test_envs.yml')
+
+    # set output
+    output_dir = os.path.join(d, 'test_out')
+
+    # prep workflow directory
+    _create_dir(output_dir)
```
Comment (qiyunzhu) [inline oecophylla/cli/launch.py]: Why not directly do `os.makedirs()` since `_create_dir` is a one-line function?

File: oecophylla/cli/launch.py [hunk 3]
```diff
@@ -49,18 +49,54 @@ def _create_dir(_path):
...
+
+def _setup_test():
+    d = _oeco_dir()
+
+    # set inputs
+    input_dir = os.path.join(d, 'test_data/test_reads')
+    sample_sheet = os.path.join(d,
+                        'test_data/test_config/example_sample_sheet.txt')
+    # set params
+    params = os.path.join(d,
```
Comment (qiyunzhu) [inline oecophylla/cli/launch.py]: if `os.path.join` looks bit to long, we can do `from os.path import join` then simply do `join`. This won't conflict with Python's native string `join`.

File: oecophylla/cli/launch.py [hunk 4]
```diff
@@ -86,67 +122,97 @@ def _create_dir(_path):
...
-    # will be automatically identified using regex.
-    if sample_sheet:
-        _sheet = read_sample_sheet(sample_sheet)
-        sample_dict = extract_samples_from_sample_sheet(_sheet, input_dir)
+    # Check to see if running with test data. If so, fill in defaults
+    # for relevant empty parameters.
+    if test:
+        input_dir, sample_sheet, params, envs, output_dir = _setup_test()
+    elif not output_dir:
+        raise IOError("Must provide output directory to run.")
```
Comment (qiyunzhu) [inline oecophylla/cli/launch.py]: Shall we uniformize double quotes into single quotes?

File: oecophylla/cli/launch.py [hunk 5]
```diff
@@ -86,67 +122,97 @@ def _create_dir(_path):
...
+        config_dict = {}
+        config_dict['samples'] = sample_dict
+        config_dict['params'] = params_dict
+        config_dict['envs'] = envs_dict
+        config_dict['tmp_dir_root'] = local_scratch
+
+        config_yaml = yaml.dump(config_dict, default_flow_style=False)
+
+
+        config_fp = '%s/%s' % (output_dir, 'config.yaml')
```
Comment (qiyunzhu) [inline oecophylla/cli/launch.py]: Though does not matter in Linux, shall we uniformize the usages of `os.path.join` and '%s/%s'?

**Review by qiyunzhu — APPROVED** [review]
Looks good. A few minor comments on code styling.

File: oecophylla/cli/launch.py (hunk 2 above)
Comment (tanaes) [reply oecophylla/cli/launch.py]: This was inherited from previous code by @mortonjt, not sure why. Presumably to allow other behavior for default directory creation?

File: oecophylla/cli/launch.py (hunk 4 above)
Comment (tanaes) [reply oecophylla/cli/launch.py]: 👍

File: oecophylla/cli/launch.py (hunk 5 above)
Comment (tanaes) [reply oecophylla/cli/launch.py]: 👍

File: oecophylla/cli/launch.py (hunk 2 above)
Comment (qiyunzhu) [reply oecophylla/cli/launch.py]: No problem!

**Comment by qiyunzhu** [comment]
@tanaes Shall we merge?

**Comment by tanaes** [comment]
would be great!

**Pull request merged**